components/ProxyFile.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace app\components;
+
+use Yii;
+use yii\base\Component;
+use yii\base\InvalidArgumentException;
+use yii\helpers\FileHelper;
+use yii\helpers\StringHelper;
+use yii\helpers\Url;
+use yii\validators\UrlValidator;
+use yii\web\Response;
+
+class ProxyFile extends Component
+{
+    /**
+     * @var string
+     */
+    public $secretKey;
+    /**
+     * @var string
+     */
+    public $storagePath = '@runtime/storageProxyFile';
+    /**
+     * @var string
+     */
+    public $storageUrl = '/storageProxyFile';
+
+    public function init()
+    {
+        if ($this->secretKey === null) {
+            $this->secretKey = Yii::$app->request->cookieValidationKey;
+        }
+
+        if (empty($this->secretKey)) {
+            throw new InvalidArgumentException('SecretKey is empty.');
+        }
+
+        parent::init();
+    }
+
+    /**
+     * @param string $url
+     *
+     * @return mixed
+     */
+    public function getConvertUrl($url)
+    {
+        if (!$this->validateUrl($url)) {
+             return $url;
+        }
+
+        $data = \Yii::$app->security->hashData($url, $this->secretKey);
+        $data = base64_encode($data);
+        $url = Url::to(['/site/proxy-file', 'data' => $data]);
+
+        return $url;
+    }
+
+    /**
+     * @param string $url
+     *
+     * @return bool
+     */
+    public function validateUrl($url)
+    {
+        if (
+            (new UrlValidator(['skipOnEmpty' => false]))->validate($url) &&
+            StringHelper::startsWith($url, 'http://') &&
+            !StringHelper::startsWith($url, Yii::$app->params['siteAbsoluteUrl'])
+        ) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $data
+     *
+     * @return Response|bool
+     */
+    public function sendFile($data)
+    {
+        $fileUrl = \Yii::$app->security->validateData(base64_decode($data), $this->secretKey);
+        if ($fileUrl === false) {
+            return false;
+        }
+
+        if (!$this->validateUrl($fileUrl)) {
+            return false;
+        }
+
+        $fileData = null;
+        try {
+            $fileData = file_get_contents($fileUrl);
+        } catch (\Exception $ex) {}
```

Comment:
If the idea to ignore exception, always add a comment stating so:

```php
} catch (\Exception $ex) {
    // ignore exception
}
```

Reply:
Done.